index.test-d.ts

```diff
@@ -8,30 +8,69 @@ import { Writable } from 'stream';
 import { err, req, res } from 'pino-std-serializers';
 import pinoHttp, { AutoLoggingOptions, CustomAttributeKeys, GenReqId, HttpLogger, Options, ReqId, startTime, StdSerializers, StdSerializedResults } from '.';
 
+interface CustomRequest extends IncomingMessage {
+  context: number;
+}
+
+interface CustomResponse extends ServerResponse {
+  context: number;
+}
+
 const logger = pino();
 
 pinoHttp();
 pinoHttp({ logger });
 pinoHttp({ logger }).logger = logger;
+pinoHttp<CustomRequest, CustomResponse>({ logger });
+
+// #genReqId
```

Comment:
The generic tests I added added a lot of clutter to this file so I've broken these option tests visually up a bit so it's easier to grok.